src/endpoints/Assets.php

```diff
@@ -16,28 +14,25 @@ public function __invoke(Request $request, Response $response)
         $service = new AssetService($this->container);
         $fileId=$request->getAttribute('id');
     
-        $response=$service->getAsset(
+        $asset=$service->getAsset(
             $fileId,
             $request->getQueryParams()
         );
         
-        if(isset($response['file']) && $response['mimeType'])
+        if(isset($asset['file']) && $asset['mimeType'])
         {
-            header('HTTP/1.1 200 OK');
-            header('Content-type: ' . $response['mimeType']);
-            header("Pragma: cache");
-            header("Content-Disposition: filename=".$response['filename']);
-            header("Access-Control-Allow-Origin: *");
-            header("Access-Control-Allow-Methods: PUT, GET, POST, DELETE, OPTIONS");
-            header("Access-Control-Allow-Headers: Access-Control-Allow-Headers,Content-Type");
-            header('Last-Modified: ' . gmdate('D, d M Y H:i:s \G\M\T', time()));
+            $response->setHeader('Content-type',$asset['mimeType']);
+            $response->setHeader('Content-Disposition','filename='.$asset['fileNameDownlaod']);
+            $response->setHeader('Last-Modified',gmdate('D, d M Y H:i:s \G\M\T', time()));    
```

Comment:
The `Last-Modified` header controls whether or not the browser will refresh its cache. By setting the header to the current time every request, the browser will never cache any thumbnail. Read the last modified date from disk and use that instead. (I already built that on the thumbnailer-cleanup branch)